### PR TITLE
feat(menu): support media type for menu

### PR DESCRIPTION
**Pull request by CarinaDraganJW**
## Description

- Support `media` type for menu items 

### Steps completed:

According to our definition of done, I have completed the following steps:

- [ ] Acceptance criteria met
- [ ] Unit tests added
- [ ] Docs updated (including config and env variables)
- [ ] Translations added
- [ ] UX tested
- [ ] Browsers / platforms tested
- [ ] Rebased & ready to merge without conflicts
- [ ] Reviewed own code

### CODE REVIEW

File: packages/common/types/config.ts [hunk 1]
```diff
@@ -44,13 +44,13 @@ export type Drm = {
 };
 
 export type PlaylistType = keyof typeof PLAYLIST_TYPE;
-
-export type PlaylistMenuType = Extract<PlaylistType, 'playlist' | 'content_list'>;
+export type PlaylistMenuType = Extract<PlaylistType, 'playlist' | 'content_list' | 'media'>;
```
Comment (AntonLantukh) [inline packages/common/types/config.ts]: Well, it looks like it is not only about playlists now. I would suggest the following:

```
PLAYLIST_TYPE -> APP_CONTENT_TYPE
PlaylistType -> AppContentType
PlaylistMenuType -> AppMenuType
PlaylistShelfType -> AppShelfType
```

File: packages/common/src/utils/urlFormatting.ts [hunk 2]
```diff
@@ -117,6 +117,10 @@ export const mediaURL = ({
   );
 };
 
+export const singleMediaURL = (id: string, title?: string) => {
```
Comment (AntonLantukh) [inline packages/common/src/utils/urlFormatting.ts:123]: @ChristiaanScheermeijer I think it makes sense to rewrite `mediaURL` function to accept id and title instead of the whole media item. Though I also see that it is used in many places, so I would prefer to do it in a separate PR.

File: packages/ui-react/src/containers/Layout/Layout.tsx [hunk 3]
```diff
@@ -24,6 +25,17 @@ import HeaderUserMenu from '../HeaderUserMenu/HeaderUserMenu';
 
 import styles from './Layout.module.scss';
 
+const determinePath = (type: PlaylistMenuType | undefined, contentId: string) => {
```
Comment (AntonLantukh) [inline packages/ui-react/src/containers/Layout/Layout.tsx]: Can we accept arguments as an object here?

File: packages/common/src/utils/urlFormatting.ts (hunk 2 above)
Comment (ChristiaanScheermeijer) [reply packages/common/src/utils/urlFormatting.ts:123]: Sounds good!

Comment (ChristiaanScheermeijer) [reply packages/common/src/utils/urlFormatting.ts:123]: Alternatively, for now you could do this as well to prevent refactoring:

```tsx
export const mediaURL = ({
  media,
  id,
  title,
  playlistId,
  play = false,
  episodeId,
}: ParamsWithMedia | ParamsWithId) => {
  const mediaId = media ? media.mediaid : id;
  const mediaTitle = media ? media.title : title;

  // ...
}
```

File: packages/common/src/utils/urlFormatting.ts [hunk 4]
```diff
@@ -125,6 +132,17 @@ export const contentListURL = (id: string, title?: string) => {
   return createPath(PATH_CONTENT_LIST, { id, title: title ? slugify(title) : undefined });
 };
 
+export const determinePath = ({ type, contentId }: { type: AppMenuType | undefined; contentId: string }) => {
+  switch (type) {
+    case APP_CONTENT_TYPE.content_list:
+      return contentListURL(contentId);
+    case APP_CONTENT_TYPE.media:
+      return singleMediaURL(contentId);
+    default:
```
Comment (AntonLantukh) [inline packages/common/src/utils/urlFormatting.ts]: We should probably check playlist explicitly, as it is still possible we get something extra in the future.

File: packages/common/src/utils/urlFormatting.ts [hunk 5]
```diff
@@ -125,6 +132,19 @@ export const contentListURL = (id: string, title?: string) => {
...
+export const determinePath = ({ type, contentId }: { type: AppMenuType | undefined; contentId: string }) => {
+  switch (type) {
+    case APP_CONTENT_TYPE.content_list:
+      return contentListURL(contentId);
+    case APP_CONTENT_TYPE.media:
+      return singleMediaURL(contentId);
+    case APP_CONTENT_TYPE.playlist:
+      return playlistURL(contentId);
+    default:
+      return playlistURL(contentId);
```
Comment (AntonLantukh) [inline packages/common/src/utils/urlFormatting.ts]:
```suggestion
      return ''
```

**Review by AntonLantukh — APPROVED** [review]

File: packages/common/src/utils/urlFormatting.ts (hunk 2 above)
Comment (CarinaDraganJW) [reply packages/common/src/utils/urlFormatting.ts:123]: I created a separate ticket: https://jwplayer.atlassian.net/browse/OWA-89 🚀
I don't know if Christiaan can see it in Jira, but we'll tackle this in a separate PR 🙂

File: packages/common/src/constants.ts [hunk 6]
```diff
@@ -86,9 +86,10 @@ export const EPG_TYPE = {
   viewNexa: 'viewnexa',
 } as const;
 
-export const PLAYLIST_TYPE = {
+export const APP_CONTENT_TYPE = {
```
Comment (ChristiaanScheermeijer) [inline packages/common/src/constants.ts]: Should we use the term "content type" here? It might be confusing with the `contentTypes` feature 🤔

Comment (AntonLantukh) [reply packages/common/src/constants.ts]: Well technically the property is `contentType`. We could use something like `APP_CONFIG_CONTENT_TYPE` or `APP_ITEM_TYPE`.

Comment (CarinaDraganJW) [reply packages/common/src/constants.ts]: Maybe we could combine the ideas and use `APP_CONFIG_ITEM_TYPE`? 🙂

Comment (AntonLantukh) [reply packages/common/src/constants.ts]: Works for me as well.

**Review by ChristiaanScheermeijer — APPROVED** [review]
Nice! 🎉
